### ISSUE TITLE
ERROR open \settings.txt:

### ISSUE THREAD

**Issue by megant** [coreybutler/nvm-windows#22]
After installation process I just ran "npm" and it says:

ERROR open \settings.txt: [...]

**Comment by jimbriam** [comment]
Also, ran into this problem, when running the 1.0.6 installer. Windows 8, Ran the installer as Admin. file.exe was not in my path. Install of GnuWin32 also did not solve the problem. Node had not been installed yet.

**Comment by kaibadash** [comment]
I could use to copy settings.txt to c:\ .

**Comment by tepez** [comment]
@bobbycircle solution worked for me.

**Comment by Schurke** [comment]
I encountered this issue also (windows 8.1). Closing and restarting the command prompt didn't help (process may still have been running - didn't check). On a hunch, I restarted windows. Problem resolved: no mucking around with the installer PATH vars. Give it a shot before changing anything 
@coreybutler - may also be a good failsafe for the installer: trigger a restart at the end "for your changes to take effect" to eliminate some of the user environment variances.

**Comment by romualdr** [comment]
Sometimes, when tweaking environnement variables, this can happen :+1: 

* You didn't push "OK" button on the "Edit environnement variable" window
Yup. This might look dumb, but i actually spent a lot of time, changing vars and restarting PowerShell, and nothing was happening ... until i closed this window ! Clicking the "OK" button is VERY important, if you close the window or cancel, your modifications will be discarded. Then just reload your term.

* You term can be faulty, try another one.
If you are using Github for Windows and Git, you might end up with 3 terms (PowerShell, Git Shell, Git Bash). Each term can have his own environnement, and tweaking values will not solve anything. Try PowerShell or cmd.exe (vanilla windows terms) before.

**Comment by bartread** [comment]
Just ran into this on Windows 8 as well.

**Comment by zonplm** [comment]
run your cmd terminal as adminstrator

**Comment by PureKrome** [comment]
Is it possible that this failed because the install failed .. because of a space in the `User` folder? This occurs in bug/issue #41 so maybe that's happening here, too?

**Comment by vineelp** [comment]
@queckezz solution worked

**Comment by matthieu-rolland** [comment]
@queckezz solution worked. thank u

**Comment by PyroSA** [comment]
I had this issue.

The installer ran with elevated privileges (required for installation).
The cmd session it launched then also runs with elevated privileges, but refused to see the settings .txt

Relaunching a non-admin cmd saw settings.txt just fine, and nvm now also runs fine in a admin session launched as an admin.

**Comment by arizonatribe** [comment]
@queckezz solution also worked for me. Thanks

**Comment by wch4896** [comment]
@queckezz solution worked +1

**Comment by giggio** [comment]
I had the same problem. The issue was that "C:\Program Files\nodejs" was in my user path, right before the new vars ;%NVM_HOME%;%NVM_SYMLINK%.

**Comment by sanori** [comment]
In my case, this phenomenon occurs just after the installation.
If I reboot after installation, nvm for windows works normally

**Comment by mthebert-mmd** [comment]
+1 for @sanori and rebooting after install fixing

**Comment by snappiness** [comment]
The problem is that PATH was not correctly updated, meaning that in my case I found that PATH value was `...";%NVM_HOME%;%NVM_SYMLINK%`. Those `...` are the existing entries, but a double quotes was added before semi colon.

Once this was corrected, `PATH = ...;%NVM_HOME%;%NVM_SYMLINK%` everything works correctly.
You don't have to copy settings.txt in C:, D: or whatever drive you are using when running nvm tool.

**Comment by PyroSA** [comment]
I also noticed the %vars% in my path, but by then had reverted to just having the active version of Node installed.

I can't believe they (MicroSoft) haven't improved the systems properties method of modifying the path. I'm sure it's been the same narrow textbox since XP days.

**Comment by giggio** [comment]
Setting env vars on Windows works just fine. You can even do it from powershell:

````powershell
[System.Environment]::SetEnvironmentVariable("MYVAR", "WHATEVER", "Machine")
````

**Comment by synthecypher** [comment]
I fixed this issue by going to `%AppData%/nvm% and changing the security of the `settings.txt` file so `Everyone` had full access to the file.

If you'd prefer not to do this you could also try running `nvm` from command prompt with elevated privileges. Using run as administrator.

**Comment by jsheely** [comment]
After it didn't work from the initially loaded "Run Cmd". I loaded an administrator cmd and ran **nvm list** which worked. Then when I went back to a regular cmd window it continued to work.

**Comment by vbdoug** [comment]
I got this error when I installed the first time. Then I ran the installer as administrator, all is well.

**Comment by mlekena** [comment]
I installed is and all that needed doing was to restart windows. I get the sense that cmd just needed to get reset with the new Environment variables.

**Comment by muthuridennis** [comment]
Same issue here. I followed @queckezz's solution, it works.

**Comment by danjah** [comment]
+1 restart on Win7

**Comment by mcpDESIGNS** [comment]
FWIW:

Just close all your instances of CMD prompt and re-open as Administrator (just in case).
Fixed it for me

**Comment by olegkalyta** [comment]
@mcpDESIGNS thank you, it helped me too!

**Comment by AdrianoCahete** [comment]
I have the same problem in Win10 x64.
But in my case is because the symbolic link pointed to a unexisting folder in C (C:\Program Files\nodejs\v1.1.2).
The only folder in nodejs folder is "v4.0.0", so i needed to remove symlink and create another using ```mklink /j "C:\Program Files\nodejs" "G:\Program Files (x86)\nvm```

**Comment by vferko** [comment]
This is still an issue on Windows 7. The variables are not being evaluated in a child cmd.

# Solution is to open a fresh cmd window.

@coreybutler I suppose you are using `SETX VAR value` and `start cmd`. 

`SETX VAR value` sets environmental variable named `VAR` for fresh instances of cmd only
`start cmd` starts a children of current cmd that uses parent's environment

A workaround is to supplement the `SETX` with `SET VAR=value` to duplicate the value for children processes.

**Comment by TKiefer** [comment]
I tried many of the above solutions, and the only one that ended up working for me was to restart Windows 7.  :disappointed:

**Comment by coreybutler** [comment]
FYI: Windows 7 support = deprecated.

**Comment by coreybutler** [comment]
@vferko - I'm not explicitly using `SET` or `SETX`. The InnoSetup engine handles all of this. I believe the problem is in sequencing. Sporadically, the installer appears to be launching the cmd window before the environment variable setup methods are complete.

**Comment by DCorto** [comment]
FYI: The ERROR open \settings.txt: [...] happened to me on a Win10 x64. After open a new cmd window seems there is all ok :)

**Comment by SleeplessByte** [comment]
Still happens with the 1.1.0 in Win10 x64. Same as @DCorto 

restart cmd.exe and you're good

**Comment by kukuruznik** [comment]
I've got somehow different symptoms, but it seems to be related to this issue:
- Win7 64bit, installation successful, but 'nvm' is not recognized as an internal or external ...
- After every system restart, nvm can be run from the command line, until the first nvm use <version> - then again 'nvm' is not recognized ...
- Deleting the User variables NVM_HOME and NVM_SYMLINK solved the issue (%vars% replaced in PATH with their values)

**Comment by coreybutler** [comment]
This should be resolved with the new installer. InnoSetup was upgraded for v1.1.7 and several new changes were made to install.cmd. Windows 7 is also not technically supported, by NVM4W or Microsoft.

**Comment by mandaputtra** [comment]
I got same error with the new 1.1.7 where is the user variable i dont see it on `settings.txt`

**Comment by mandaputtra** [comment]
Sorry just restart the computer and its work 💯

**Comment by TomasHubelbauer** [comment]
This error is still present. I was using NVM without an issue for a while and now I've had a project which required me to move my NPM cache directly to the C drive (to avoid long path issues) and after I've made this change (probably, that's the only NPM configuration change that I did and the same day this started happening…) this issue started happening.

I found that running Powershell as an admin allow me to run NVM with no issues, which is a workaround I'll be using for now (I might set the `settings.txt` file permissions to "anything goes" like the other poster if it gets too annoying).

**Comment by toddalbers** [comment]
I don't know think this will be a fix for all.  SHORT ANSWER:  Try closing the command window and opening a new command window.  LONG ANSWER:  I found that if I installed nvm using the nvm_setup.exe (downloaded in nvm_setup.zip) from the windows command line (cmd.exe command window) and then tried to run nvm in the same command window session (right after installing), it gave me this error -   ERROR open \settings.txt.  I looked and the paths were added to my system PATH settings fine.  So, I closed the command line window and opened up a new command window.  It then worked.  I think even though the install adds the path to nvm.exe (and to settings.txt) to the Windows PATH, it isn't recognized in that same command window yet.  Close the command window and open a new one.  Hope this helps someone.

**Comment by Muizuddeen** [comment]
I got error when i ran npx create react app it says enter the exact path where the zip file is extracted/copied: it's having after nvm installation and installation of node 12.

**Comment by SleeplessByte** [comment]
> Still happens with the 1.1.0 in Win10 x64. Same as dcorto
> 
> restart cmd.exe and you're good

This stil works (as workaround) @Muizuddeen

**Comment by AdrianoCahete** [comment]
Found that if you copy the `settings.txt` to `c:` works too

**Comment by dylanh724** [comment]
This issue still persists in Win11 if you install via Choco (in admin PS window). Let's say you have a giant install script -- it halts here since refreshing the env via powershell does NOT seem to resolve the issues. 

Restarting PS fixes it, but this is breaking for silent install scripts. Any admin powershell cmd(s) I can use to resolve this without restarting so I can carry on my epic silent install script for fresh Windows installs? :D

**Comment by emallove** [comment]
I hit this issue on [v1.1.9](https://github.com/coreybutler/nvm-windows/releases/tag/1.1.9.).

**Comment by DigDugIA** [comment]
Greetings.  Thanks for all your time supporting NVM.   

I just downloaded and ran the install EXE for NVM 1.1.10.  I do the nvm -v and get that version number back.  I do nvm by itself and get the help info.   When I try the "nvm install latest" command I get this back. 

ERROR open \settings.txt: The system cannot find the file specified.

I can see all the files listed at the top of this Issue.  

Directory Image:  
![image](https://user-images.githubusercontent.com/14084100/218593353-24c64a95-c79f-45fc-9901-1ebceb1ec62c.png)

Not sure what to do next.  Need to get NPN installed, so I can run a OpenAPI generator to get a C# API library for a company I need to do API calls into.  

Thanks.

**Comment by chill389cc** [comment]
I get this error sporadically on windows, which is annoying. I know that there are no issues with my path because I can enter  `settings.txt` in a shell and it opens up the nvm settings. I'm not sure why nvm can't open it sometimes if my shell has no issues.

**Comment by jimtut** [comment]
Just got this with 1.1.11 today.  The error is very specific that "\settings.txt" doesn't work.  You need to put the file at the root of the hard-drive.  Not sure why, but even when running from c:\nvm\, it couldn't read the settings.txt from there, needed to be at c:\settings.txt.

**Comment by JackAllread** [comment]
Above it was stated that the **settings.txt** file's content was:

   **root**: C:\Users\Corey\AppData\Roaming\nvm
   **path**: C:\Program Files\nodejs

But, I found that it's the other way around:

   **path**: C:\Users\Corey\AppData\Roaming\nvm
   **root**: C:\Program Files\nodejs

And with the security on my dev computer as tight as it was I had to place the nvm folder in 
   **path**: C:\Users\Public\Roaming\nvm

Which worked fine.

**Comment by rushian** [comment]
> Where the installed application goes... the default for this is `C:\Users\<yourusername>\AppData\Roaming\nvm`. It should look something like this:
> 
> ![image](https://cloud.githubusercontent.com/assets/770982/5145723/f6b1e0a0-716a-11e4-9c54-e95a7389322f.png)
> 
> The main files to look for are `elevate.cmd`, `elevate.vbs`, `nvm.exe`, and `settings.txt`, as well as `nodejs.ico` (which is just the icon, it doesn't provide any functionality).
> 
> If `settings.txt` does not exist, then it probably wasn't created during the installation process. The installer should do this for you, but the manual install does not. If you used the installer and do not see the `settings.txt` file, then there is probably a bug in the installer that I'd be interested in tracking down and fixing.
> 
> The settings file should contain content similar to:
> 
> ```
> root: C:\Users\Corey\AppData\Roaming\nvm
> path: C:\Program Files\nodejs
> ```

thank you... I used the zip file to use nvm... there is not this file in zip not intallable... then I created it with this information and did work...

**Comment by kiraio-moe** [comment]
## Quick Fix

- Go to nvm-windows installation.
- Run `install.cmd` as Admin. (Optional)
- Open `settings.txt` and set your current nvm-windows and nodejs paths:

  ```
  root: C:\Programs\nvm
  path: C:\Programs\nodejs
  arch: 64
  proxy: none
  ```

- Copy `settings.txt` to `C:` drive.
- Open your Environment Variables and set the `NVM_HOME` & `NVM_SYMLINK` paths same as in `settings.txt`:

  ![nvm-windows_env-settings](https://github.com/coreybutler/nvm-windows/assets/58289710/7a4b5df7-1028-4da3-980d-b6cc0685d324)

**Comment by kiraio-moe** [comment]
## Quick Fix

- Go to nvm-windows installation.
- Run `install.cmd` as Admin. (Optional)
- Open `settings.txt` and set your current nvm-windows and nodejs paths:

  ```
  root: C:\Programs\nvm
  path: C:\Programs\nodejs
  arch: 64
  proxy: none
  ```

- Copy `settings.txt` to `C:` drive.
- Open your Environment Variables and set the `NVM_HOME` & `NVM_SYMLINK` paths same as in `settings.txt`:

  ![nvm-windows_env-settings](https://github.com/coreybutler/nvm-windows/assets/58289710/7a4b5df7-1028-4da3-980d-b6cc0685d324)

**Comment by balaji-b-v** [comment]
I ran into the same problem when running **nvm no-install** fileset https://github.com/coreybutler/nvm-windows/releases/download/1.1.12/nvm-noinstall.zip

Steps to resolve: (When you do not have Admin permission on your system)
1. Create two folders nvm, and nodejs at C:\ 
![image](https://github.com/coreybutler/nvm-windows/assets/3445541/71e2549c-450a-4e7e-96b4-2069523d497e)
2. Unzip the contents of above URL into nvm folder 
![image](https://github.com/coreybutler/nvm-windows/assets/3445541/8152a9a6-2252-420f-a00d-f5f4de3621de)
3. Create a settings.txt file with below contents
`root: C:\nvm
path: C:\nodejs
arch: 64
proxy: none`
4. Open command prompt
5. move to C:\nvm folder
6. Run cmd `nvm list` to see output
![image](https://github.com/coreybutler/nvm-windows/assets/3445541/71e96690-94a8-4bb2-82e8-4146fcc16101)
7. The run the cmd to install your chosen version eg: `nvm install 20.10.0`  to see
![image](https://github.com/coreybutler/nvm-windows/assets/3445541/3f2860cf-fc06-49fe-8d59-0489fe7f06c5)

**Comment by jefflomax** [comment]
Also ran into this on Windows 10 22H2 19045.4291
Moving the settings.txt file to c:\ works - otherwise the very specific error opening settings.txt
LONG SURVIVING BUG

**Comment by Cifko** [comment]
Still an issue as of today.
Also when I copy it to C: and I start powershell on D and run the nvm, it's an issue. So it basically looks up for the file in the root of the current drive.

**Comment by nickolasdeluca** [comment]
I don't think this will get fixed. I believe that ever since Corey started working full time on `Runtime`, nvm-windows has been in maintenance mode only. And to be honest, this is probably not worth spending time to fix. But yeah, still an issue.

In my case, I was automating the installation of some dependencies, one of them was nvm. Since I'm installing it manually, I had to reload the Powershell environment variables manually to make the script be able to use nvm without reloading PS.

Simply reloading `$env:Path` is not enough, you also have to manually reload `$env:NVM_HOME` and `$env:NVM_SYMLINK`.

```powershell
$env:Path = [System.Environment]::ExpandEnvironmentVariables(([System.Environment]::GetEnvironmentVariable("Path", "Machine") + ";" + [System.Environment]::GetEnvironmentVariable("Path", "User")))
$env:NVM_HOME = [System.Environment]::GetEnvironmentVariable("NVM_HOME", "User")
$env:NVM_SYMLINK = [System.Environment]::GetEnvironmentVariable("NVM_SYMLINK", "User")
```

Basically, those three lines together solved the issue.

Oh and if you're installing it manually, make sure the `NVM_SYMLINK` folder exists, otherwise when you use the `use` command, it will not create the symlink.

If `NVM_SYMLINK` is `C:\nvm4w\nodejs`, you have to make sure that `C:\nvm4w` exists.
